gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -51,9 +70,20 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MysqlDatasetStateStore.class);
 
+  private static final String SELECT_JOB_STATE_LATEST_BY_PREFIX_TEMPLATE =
+          "SELECT state, CASE WHEN locate(?, table_name) = 0 THEN '' "
```

Comment:
I discussed with @ibuenros and he said there have been discussions about reorganizing the dataset state into subdirectories. That would remove the need for parsing. He also raised concerns about the cost of listing vs being able to read the file directly.

One suggestions to avoid and backward compatibility and performance issue is to support both 'alias-based' and 'time-based' state stores either by abstracting out a new state store type or having it based on configuration within the existing state stores. That approach would leave createAlias() and use it if configured.